test/e2e2/spec/test_dyndns.py

```diff
@@ -41,16 +41,21 @@ def do_test(url, headers, params, expected_ipv4, expected_ipv6, subname):
             }
             for qtype in ['A', 'AAAA']
         }
-        rrs_dns = {
-            qtype: {rr.to_text() for rr in NSLordClient.query(params.get('hostname', domain), qtype) or []}
-            for qtype in ['A', 'AAAA']
-        }
 
         for expected_net, qtype in [(expected_ipv4, 'A'), (expected_ipv6, 'AAAA')]:
             assert len(rrs_api[qtype]) == (1 if expected_net else 0)
-            assert len(rrs_dns[qtype]) == (1 if expected_net else 0)
             assert _ips_in_network(rrs_api[qtype], expected_net)
-            assert _ips_in_network(rrs_dns[qtype], expected_net)
+
+            def assertion(query):
+                rrs_dns = {
```

Comment:
No because the call is also inside the for loop:

```shell
>>> for i in range(2):
...   def callme():
...     print(i)
...   callme()
... 
0
1
```

Reply:
Oh yes. I had written that assuming that you would fix then typo `qtype` to `qt`, which would then make the call independent of the iteration variables. But you refactored, so no longer applies.